deidentification/utils/get_tags_diff.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import difflib
+import pydicom
+import os, re
+import argparse
+
+def get_tag_diff(path_raw, path_ano, outpath=None):
+    """
+    Return diff.txt file in the sam directory as path_raw which
+    contain the tag difference between the 2 dcm 
+
+    Args:
+        path_dcm1 (str): path of the first dcm (ref)
+        path_dcm2 (str): path of the second dcm (current)
+        outpath (str): path of the diff_file
+    """
+    print(f"Check difference between :\n{path_raw} and\n{path_ano}\n")
+    dcm_ref = pydicom.read_file(path_raw)
+    dcm_current = pydicom.read_file(path_ano)
+    diff = difflib.ndiff(str(dcm_ref).splitlines(keepends=True),
+                         str(dcm_current).splitlines(keepends=True))
+    
+    diff_from_dcm1 = []
+    diff_from_dcm_2 = []
+    global_modif = []
+    current_line_diff = None
+    previous_tag = None
+    previous_sign =  None
+
+    for i in diff:
+        if "(" not in i or ")" not in i:
+            continue
+        tag = i[i.index('('): i.index(')')+1]
+        txt = ""
+
+        # tag present in only one dcm
+        if tag != previous_tag:
+            if i[0] == "-":
+                txt = "Dcm1 only:"
+                current_line_diff = diff_from_dcm1
+            elif i[0] == "+":
+                txt = "Dcm2 only:"
+                current_line_diff = diff_from_dcm_2
+            line = txt + i[1:]
+        # between 2 diff lines : same tag + remove from first line and add to second -> value modification
+        elif tag == previous_tag and i[0] != previous_sign:
+            line = ""
+            txt = f"Modification (dcm1 -> dcm2): {tag}"
+            previous_elem = current_line_diff.pop(-1)
+            # check tag name
+            previous_name = previous_elem[previous_elem.index(')') + 1: previous_elem.index(re.search(' [A-Z]{2}:', previous_elem).group())].replace("  ", "")
+            current_name = i[i.index(')') + 1: i.index(re.search(' [A-Z]{2}:', i).group())].replace("  ", "")
```

Comment:
It seems a little bit complicated. You can try to use regex to catch all the elements in your line.
```suggestion
            # \([0-9a-f]{4}, [0-9a-f]{4}\) : find the tag
            # (.*\S+) : catch the Name of the tag
            # [A-Z]{2}: : find the type of value
            # (.*)$ : catch the Value of the tag
            previous_name, previous_value = re.search(r'^.*\([0-9a-f]{4}, [0-9a-f]{4}\) (.*\S+).*[A-Z]{2}: (.*)$',
                                                      previous_elem).groups()
            current_name, current_value = re.search(r'^[+-] \([0-9a-f]{4}, [0-9a-f]{4}\) (.*\S+).*[A-Z]{2}: (.*)$',
                                                    i).groups()
```

Then you can catch several values from your regex. Do not hesitate to use site like [regex101](https://regex101.com/) to understand and create regex :wink:

Reply:
Then you can also adapt the regex used in `i` to also catch the sign and the complete tag earlier in the code if you want.

Reply:
I did use regex, I will try simplify them